### ISSUE TITLE
Fan Speed not working also Turn off/on function not working

### ISSUE THREAD

**Issue by Axelf-alex** [hunterjm/ac-infinity-hacs#9]
Hello @hunterjm 

Thanks for the amazing integration but having some issues. I have a 69 BLE only Controller Home Assistant Recognised the device gives me correct temparature/humidity reading and also when fan is running via the app it recognises its running and speed. 

The issue is when trying to controller it via the integration it's not responding. I have turned off bluetooth on my phone and deleted any in app automatiosn but no luck on that. When I try to run the fan via Home Assistant its saying failed to update or not making any changes at all. For example if the fan is off in app: will show off in HA but then I try to turn it ON in HA and flips on its own back off and vice versa. With the speed its the same so the problem is that its reading the state data but fails to push the speed and state and update the controller. 

I am happy to provide any logs but will need some help on where to grab them. 

Thank you in advance

**Comment by dagrah** [comment]
For me it works this way:
Set your controller to ON (via app or directly on the controller)
Close/Kill the app(task) on your phone
At this moment your fan should interact with your HA Commands.

**Comment by Axelf-alex** [comment]
Unfortunately I tried this too and just tried it just in case but it’s not working. Don’t know if there is an issue if the BLE adapter is a ESP32, if there was an issue I believe nothing would work, which in this case I have some functionality. Thank you for your advise tho.

**Comment by tsaro** [comment]
I'm having the same issue here with btle gateways via esphome, getting temp/humidity telemetry from the unit and can see if its on/off and the speed but I get an error in HA,  "Failed to call service fan/turn_off  unknown error. "

**Comment by flamedodger** [comment]
> I'm having the same issue here with btle gateways via esphome, getting temp/humidity telemetry from the unit and can see if its on/off and the speed but I get an error in HA, "Failed to call service fan/turn_off unknown error. "

Have you turned your BT to active scanning?

**Comment by tsaro** [comment]
> Have you turned your BT to active scanning?

Thanks for the quick reply, I am using active scanning, I did notice that the bluetooth icon flashes on and off repeated when I'm testing, I do have multiple esp devices with gateway enabled in range of this device, I wonder if that could be a factor. 
```
esp32_ble_tracker:
  scan_parameters:
    active: true

bluetooth_proxy:
  active: true
```

**Comment by Thund3r1** [comment]
> > Have you turned your BT to active scanning?
> 
> Thanks for the quick reply, I am using active scanning, I did notice that the bluetooth icon flashes on and off repeated when I'm testing, I do have multiple esp devices with gateway enabled in range of this device, I wonder if that could be a factor.
> 
> ```
> esp32_ble_tracker:
>   scan_parameters:
>     active: true
> 
> bluetooth_proxy:
>   active: true
> ```

Having the same issue, did you ever find a fix?

**Comment by onefoursixteen** [comment]
The ac-infinity-ble library is hard coded to port 0. I  had to select all port form the app(most probably port 0) and now all the functions are working.

**Comment by Axelf-alex** [comment]
@onefoursixteen  how did you manage to do this? Any instructions will be highly appreciated. Thank you

**Comment by onefoursixteen** [comment]
@Axelf-alex I have just selected All in the app 
![Screenshot_20260403-204819.png](https://github.com/user-attachments/assets/49dffdc2-53ed-41d8-bd9b-6b42db6c34f5)

**Comment by Axelf-alex** [comment]
@onefoursixteen  Thank you for this. Do you use ESP32-Home as a bluetooth proxy ?

**Comment by onefoursixteen** [comment]
@Axelf-alex  Yes. I am using M5Stack ESP32 PoE Ethernet with ESP Home as Bluetooth proxy.

**Comment by dagrah** [comment]
For me it works this way:
Use BT Proxy with any ESP32 or something (got a lot of trouble with direct BT USB device and didn´t really worked). 
If you have more fans it can be possible that you will need more BT Proxys since one Proxy only can handle 3 connections if i´m rembember correctly.  Depends on the automation, but it could be possible, that your fan will spam your BT Proxy (with multible BT Proxys you will have no problems)
Set your controller to ON (via app or directly on the controller ) -> VERY IMPORTANT
Close/Kill the app(task) on your phone (VERY IMPORTANT TOO)
At this moment your fan should interact with your HA Commands.

**Comment by Axelf-alex** [comment]
@dagrah  did you use ESPHOME or just ESP32 bluetooth proxy?

**Comment by dagrah** [comment]
I flashed it here: https://esphome.io/projects/?type=bluetooth

But this is my final config for my proxies ;)

**********************************************************************************
**********************************************************************************


esp32:
  framework:
    type: esp-idf

substitutions:
  name: bluetooth-proxy-buro
  friendly_name: Bluetooth Proxy Büro
packages:
  esphome.bluetooth-proxy: github://esphome/bluetooth-proxies/esp32-generic/esp32-generic.yaml@main
esphome:
  name: ${name}
  name_add_mac_suffix: false
  friendly_name: ${friendly_name}

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

bluetooth_proxy:
  active: true
